### PR TITLE
Week4

### PR DESCRIPTION
**Pull request by punyakriti5**

### CODE REVIEW

File: Week-3/Session7-Responsiveness/VacationRespImg/responsiveImg.html [hunk 1]
```diff
@@ -29,13 +29,19 @@ <h2>Egypt</h2>
...
-        sizes="(max-width: 500px) 200px, (min-width: 500px) 400px, (min-width: 700px) 600px"
-      />
+       
+        srcset="luxor_600.jpg 600w,
+                 luxor_800.jpg 800w,
+                 luxor_1200.jpg 1200w" 
+         sizes="(min-width: 700px) 600px,
+                (max-width: 500px) 400px,
+                200px" 
+         alt="Luxor Temple">
```
Comment (svashish305) [inline Week-3/Session7-Responsiveness/VacationRespImg/responsiveImg.html:42]: is this working now? what was the issue earlier?

**Review by svashish305 — CHANGES_REQUESTED** [review]
can you merge latest master to week4 branch first to only have week4 changes? currently previous week changes are also in this PR which shouldn't happen.

File: Week-3/Session9 Basics JS/strongPasword.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        let check;
+        
+        for (let i = 0; i < str.length; i++) {
+          if (str[i].valueOf().toUpperCase() === str[i].valueOf()){
+           uppercase = true;
+            break;
+          } else {
+            uppercase = false;
+          }
+        }
```
Comment (svashish305) [inline Week-3/Session9 Basics JS/strongPasword.html:21]: refactor to use password.some()

File: Week-4/Session 10 JS/S10_call_apply_bind.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    return a - b;
+  }
+  multiply(a, b) {
+    return a * b;
+  }
+  divide(a, b) {
+    return a / b;
+  }
+}
+class ScientificCalculator extends Calculator {
```
Comment (svashish305) [inline Week-4/Session 10 JS/S10_call_apply_bind.js:15]:
```suggestion

class ScientificCalculator extends Calculator {
```

File: Week-4/Session 10 JS/S10_this.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+class Person {
+  constructor(Name, Age, Gender, Nationality) {
```
Comment (svashish305) [inline Week-4/Session 10 JS/S10_this.js]: ideally the arguments shouldn't be capitalized. only class names in js are capitalized.

Comment (svashish305) [reply Week-4/Session 10 JS/S10_this.js]: refactor accordingly in this file.

File: Week-3/Session7-Responsiveness/VacationRespImg/responsiveImg.html (hunk 1 above)
Comment (punyakriti5) [reply Week-3/Session7-Responsiveness/VacationRespImg/responsiveImg.html:42]: > is this working now? what was the issue earlier?

yes, earlier I was providing size in px in the srcset, now its adjusting acc. to viewport

File: Week-3/Session9 Basics JS/strongPasword.html (hunk 2 above)
Comment (punyakriti5) [inline Week-3/Session9 Basics JS/strongPasword.html:21]: str=["e","E","F","f"];
let uppercase;
function isUpperCase() {
     for (let item of str) {
      if (item.valueOf().toUpperCase() === item.valueOf())
         return true;
       return false;
     }
  }
   str.some(isUpperCase) ? uppercase = true: uppercase = false;
 console.log(uppercase);


I tried using some, but its returning false for every iteration every when it is true

File: Week-4/Session 10 JS/S10_call_apply_bind.js (hunk 3 above)
Comment (punyakriti5) [reply Week-4/Session 10 JS/S10_call_apply_bind.js:15]: > can you merge latest master to week4 branch first to only have week4 changes? currently previous week changes are also in this PR which shouldn't happen.

beacause I guess I didnt merge week3 code, as it needs refactoring

File: Week-4/Session 10 JS/S10_this.js (hunk 4 above)
Comment (punyakriti5) [reply Week-4/Session 10 JS/S10_this.js]: okay

**Review by svashish305 — APPROVED** [review]

**Pull request merged**